### PR TITLE
add cors plugin option

### PR DESCRIPTION
**Pull request by philbrookes**
add cors plugin

### CODE REVIEW

**Comment by david-martin** [comment]
@philbrookes i can try this out if you have an image pushed somewhere

File: roles/provision-3scale-apb/defaults/main.yml [hunk 1]
```diff
@@ -2,4 +2,6 @@
 threescale_deploy_template: "https://raw.githubusercontent.com/3scale/3scale-amp-openshift-templates/2.0.0.GA-redhat-2/apicast-gateway/apicast.yml"
 threescale_image: "quay.io/3scale/apicast:master"
 threescale_basic_auth_secret_name: "apicast-configuration-url-secret"
-threescale_port: "8080"
\ No newline at end of file
+threescale_port: "8080"
+threescale_apicast_cors: "https://raw.githubusercontent.com/3scale/apicast/master/examples/cors/cors.lua"
```
Comment (matzew) [inline roles/provision-3scale-apb/defaults/main.yml]: just one question - should we point to tags/ instead of master branch ? As things may change ?

**Comment by matzew** [comment]
Giving it a look now ... :eyes:

**Comment by philbrookes** [comment]
@david-martin cool thanks, I have an image in docker.io/pbrookes/3scale-apb:latest

**Comment by david-martin** [comment]
@philbrookes :+1: 

Catalog item has checkbox for CORS, enabled by default.
image

After provisioning, a curl request (OPTIONS) gives a 204 and expected headers
```
curl -I -X OPTIONS -H "origin:https://192.168.37.1:8443" -H "access-control-request-headers:authorization" -H "access-control-request-method:GET" http://apicast-myproject.192.168.37.1.nip.io/sync
HTTP/1.1 204 No Content
Server: openresty/1.11.2.5
Date: Wed, 11 Oct 2017 14:56:17 GMT
Access-Control-Allow-Headers: authorization
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: https://192.168.37.1:8443
Access-Control-Allow-Credentials: true
Set-Cookie: c35df51fb9607cae23f5f59ad41c9501=3acb3a3a2d1f26bfe8a8e69c9a3bed6d; path=/; HttpOnly
```

Nitpicks
* "Enable CORS 3scale plugin?" => capitalize Scale i.e. 3Scale
* Is that a 3Scale icon? Is there a 3Scale icon available?
* I'm thinking the description on the left could give some info on what exactly gets deployed (APICast) and why a token and domain are needed. There is an onprem version of 3Scale which (I'm guessing) includes all parts of 3Scale, so requiring a Token & Domain to be specified could seem unusual at install time.

**Comment by david-martin** [comment]
@philbrookes 1 more thing (apologies).
Could we add the mobile-service category to the apb so it shows up in 'Mobile > Services' in the Service Catalog?

**Comment by philbrookes** [comment]
@david-martin made those requested changes now. It is not the 3Scale icon, but I couldn't find one that we could use, so I just picked something cool-looking :)

**Pull request merged**